### PR TITLE
add Makefile

### PR DESCRIPTION
**Pull request by gdevillele**
Started to write a Makefile for this repo.
This is still WIP, comments are welcome.

@dnephin Thanks for your Dockerfile! #1 

Signed-off-by: Gaetan de Villele <gdevillele@gmail.com>

### CODE REVIEW

**Comment by dnephin** [comment]
I think we should try and keep "targets that run inside a container" in a separate file from "targets that build images or run containers".

Anyone who wants to run the "inside a container" targets on their host can do that by pointing at the right `Makefile`.

Maybe we can have `Makefile.go` for "inside a container" targets.

File: dockerfiles/Dockerfile.build [hunk 1]
```diff
@@ -2,6 +2,7 @@
 FROM    golang:1.8-alpine
 
 RUN     apk add -U git
+RUN		apk add -U make
```
Comment (dnephin) [inline dockerfiles/Dockerfile.build]: Looks like formatting is off a bit here because of tabs vs spaces

Comment (gdevillele) [inline dockerfiles/Dockerfile.build]: Thx! On it.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#
+# github.com/docker/cli 
+#
+
+# build the cli
+build:
+	go build -o ./build/docker ./cmd/docker
```
Comment (dnephin) [inline Makefile]: To build all the binaries we can do this:

```
    gox -output dist/docker-{{.OS}}-{{.Arch}}
        -osarch="linux/arm linux/amd64 darwin/amd64 windows/amd64"
        ./cmd/docker

```

Comment (gdevillele) [inline Makefile]: Thanks! Used it for the "cross" target :)

File: dockerfiles/Dockerfile.build [hunk 3]
```diff
@@ -2,6 +2,7 @@
 FROM    golang:1.8-alpine
 
 RUN     apk add -U git
+RUN     apk add -U make
```
Comment (dnephin) [inline dockerfiles/Dockerfile.build]: This should be one line:

```
RUN   apk add -U git make
```

Comment (gdevillele) [inline dockerfiles/Dockerfile.build]: on it :)

**Pull request merged**